### PR TITLE
Add spatialScalability to VideoConfiguration

### LINKED ISSUE #182: Use referenceScaling instead of scalabilityMode for decodingInfo queries
During the implementation in Chrome I've come across an issue related to scalabilityMode.

Scalability mode specifies how the video stream should be encoded and puts a few requirements on the encoder. In theory all decoders should support decoding of all valid streams regardless of which scalability mode that was used (https://www.w3.org/TR/webrtc-svc/). In reality this is not always the case and it may happen that a decoder is not able to decode a stream due to certain properties of the stream. All cases I know of where this happens are tied to reference scaling, which means that a frame of resolution A is used as a reference when decoding a frame of resolution B != A.

My proposal is therefore to add a boolean called referenceScaling to the dictionary VideoConfiguration.

referenceScaling would be an optional member that could be set when querying decodingInfo(). The member scalabilityMode remains in the dictionary but will only be allowed when querying encodingInfo.

### PR DESCRIPTION
**Pull request by drkron**
Fixes #182

***
<a href="https://pr-preview.s3.amazonaws.com/drkron/media-capabilities/pull/189.html" title="Last updated on Feb 8, 2022, 10:36 PM UTC (ee9acf7)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/media-capabilities/189/7cfc693...drkron:ee9acf7.html" title="Last updated on Feb 8, 2022, 10:36 PM UTC (ee9acf7)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -476,8 +477,18 @@ spec: secure-contexts; urlPrefix: https://www.w3.org/TR/secure-contexts/
       <p>
         If present, the <dfn for='VideoConfiguration' dict-member>scalabilityMode</dfn>
         member represents the scalability mode as defined in [[webrtc-svc]].
+        Scalability mode is only applicable to the MediaEncodingType webrtc.
```
Comment (chcunningham) [inline index.bs:480]: nits:
* use same scalabilityMode spelling here as in dict-member. 
* make MediaEncodingType and webrtc link to their dfns in this spec.

File: index.bs [hunk 2]
```diff
@@ -476,8 +477,18 @@ spec: secure-contexts; urlPrefix: https://www.w3.org/TR/secure-contexts/
...
       </p>
 
+      <p>
+        If present, the <dfn for='VideoConfiguration' dict-member>spatialScalability</dfn>
+        member represents the ability to do spatial prediction, that is,
+        using frames of a resolution different than the current resolution as
+        dependencies. This is closely coupled to {{scalabilityMode}} in the
```
Comment (chcunningham) [inline index.bs:487]: I'd massage the phrasing of this sentence a bit for clarity. Something like:

... in the sense that streams encoded with modes using spatial scalability (e.g. "L2T1") can only be decoded if spatialScalability is supported.

File: index.bs (hunk 1 above)
Comment (chcunningham) [inline index.bs:480]: Before this sentence, can you add something like: If absent, the default mode of "L1T1" (no scalability) is assumed. 

This spec hasn't done a great job w/ documenting defaults. Trying to make it better incrementally.

File: index.bs (hunk 2 above)
Comment (chcunningham) [inline index.bs:487]: As a second sentence, can you add something like: If absent, spatialScalability will default to `false`.

File: index.bs [hunk 3]
```diff
@@ -476,8 +477,18 @@ spec: secure-contexts; urlPrefix: https://www.w3.org/TR/secure-contexts/
...
+        sense that streams that are produced using certain scalability modes
+        can only be decoded if spatalScalability is supported. Spatial
+        scalability is only applicable to MediaDecodingConfiguration.
```
Comment (chcunningham) [inline index.bs:490]: Nit: use same spelling as dict-member

**Review by chcunningham — COMMENTED** [review]
All minor feedback

File: index.bs (hunk 2 above)
Comment (drkron) [reply index.bs]: Done.

File: index.bs (hunk 1 above)
Comment (drkron) [reply index.bs]: Done.

File: index.bs (hunk 2 above)
Comment (drkron) [reply index.bs]: Done.

File: index.bs (hunk 3 above)
Comment (drkron) [reply index.bs]: Done.

**Comment by drkron** [comment]
> All minor feedback

Thanks for the feedback! I've updated the PR according to your suggestions.

**Comment by drkron** [comment]
@chcunningham 
What do you think of the current state of this PR? Do you have any other comments?

**Comment by aboba** [comment]
Is "L1T1" always the default scalability mode for every codec?  For VP8 in Chromium WebRTC, my understanding is that this is not the case.

**Comment by chcunningham** [comment]
cc @alvestrand 

> Is "L1T1" always the default scalability mode for every codec? For VP8 in Chromium WebRTC, my understanding is that this is not the case.

Very interesting, I hadn't realized. I found [more context here](https://github.com/w3c/webrtc-svc/issues/48#issuecomment-951239917)). 

Am I right that the vp8 default is entirely up to implementers? No spec that spells it out? 

We could amend this area of the spec to say something like: "the default value implementer defined choice correspsonding to the default scalabilityMode (i.e. the mode you get if you don't specify one via `setParameters()`)". It's a bit subtle, but probably less subtle than having different defaults between different APIs. 

> What do you think of the current state of this PR? Do you have any other comments?

LGTM % lingering default concern.

**Comment by chcunningham** [comment]
LGTM, but ping @aboba @alvestrand to double check the approach in that last commit.

**Review by alvestrand — APPROVED** [review]

**Comment by alvestrand** [comment]
Comment on VP8: the implementors actually changed the default from L1T3 to L1T2 some time back. There were some howls of outrage from people who felt that L1T3 worked better for them, and a renewed interest in having a control surface for it.

**Review by aboba — APPROVED** [review]
Looks fine.

**Comment by chcunningham** [comment]
Dusting this off now that I'm back from leave. Thanks for reviewing! Merging now.

**Pull request merged**